File: programs/gcd.pl
sub gcd ($$) {
   my ($x, $y) = @_;
   while ($x != $y) {
      if ($x > $y) {$x -= $y} else {$y -= $x}
   }
   return $x
}

#IFFY
sub gcd {
    my ($x, $y) = @_;
    my $big = max($x, $y);
    my $small = min($x, $y);
    while ($big % $small != 0) {
        my $temp = $big - $small;
        $big = max($temp, $small);
        $small = min($temp, $small);
    }
    return $small;
}

File: programs/gcd.ml
let rec gcd x y =
    if x > y then gcd (x - y) y
    else if x < y then gcd x (y - x)
    else x;;

*IFFY*
let gcd x y =
    let rec foo a b =
        if (a mod b == 0) then b else (let temp = a - b in foo (max temp b) (min temp b))

in foo (max x y) (min x y);
